### DISCUSSION TITLE
Composable API

### DISCUSSION THREAD

**Discussion by rizeki-rifandani** [hugocxl/react-echarts#55] [Ideas]
Hi, I just wondering is it possible, instead of passing a lot of props into one `EChart` component or `useECharts` hooks, could we compose it like this:

```tsx
<EChart>
            <CartesianGrid stroke="#f5f5f5" />
            <XAxis dataKey="name" scale="band" />
            <YAxis />
            <Tooltip />
            <Legend />
            <Area type="monotone" dataKey="amt" fill="#8884d8" stroke="#8884d8" />
            <Bar dataKey="pv" barSize={20} fill="#413ea0" />
            <Line type="monotone" dataKey="uv" stroke="#ff7300" />
</EChart>
```

This is exactly what [recharts](https://recharts.org/en-US) does which is based on [Victory](https://commerce.nearform.com/open-source/victory/docs)

**Comment by hugocxl** [comment]
Hi @rizeki-rifandani, great suggestion! Currently it's not possible, but it's actually my idea for a v2 version of the lib, though I cannot find time these days to get it done. I'm open to receive contributions that lead the lib in that direction!
